src/cache/encryption.rs

```diff
@@ -0,0 +1,26 @@
+#![allow(clippy::type_complexity)] // the static types are too long, this cases no compilation problems though.
+use chacha20poly1305::{
+    consts::{B0, B1},
+    ChaChaPoly1305,
+};
+use std::sync::OnceLock;
+
+use chacha20::{
+    cipher::{
+        generic_array::GenericArray,
+        typenum::{UInt, UTerm},
+        StreamCipherCoreWrapper,
+    },
+    ChaChaCore,
+};
+
+/// OUR CIPHER, only initialised once
+pub static CIPHER: OnceLock<
+    ChaChaPoly1305<
+        StreamCipherCoreWrapper<ChaChaCore<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>>>,
```

Comment:
```suggestion
/// --- Add appropriate documentation for this type ---
type ChaChaCoreStreamCipher = ChaChaPoly1305<StreamCipherCoreWrapper<ChaChaCore<UInt<UInt<UInt<UInt<UTerm, B1>, B0>, B1>, B0>>>;

use chacha20poly1305::{
    consts::{B0, B1},
    ChaChaPoly1305,
};
use std::sync::OnceLock;

use chacha20::{
    cipher::{
        generic_array::GenericArray,
        typenum::{UInt, UTerm},
        StreamCipherCoreWrapper,
    },
    ChaChaCore,
};

/// OUR CIPHER, only initialised once
pub static CIPHER: OnceLock<ChaChaCoreStreamCipher>,
```

I think instead of allowing this Clippy lint, I would suggest instead defining a custom type and just replace it with it in the code. :slightly_smiling_face: